pkg/metrics/metrics.go

```diff
@@ -57,26 +58,38 @@ func (m *metricsFactory) getCurryLabels(labels []string) ([]string, prometheus.L
 
 func (m *metricsFactory) NewCounter(name string, help string, labels []string) *prometheus.CounterVec {
 	labels, cl := m.getCurryLabels(labels)
+	ns := "kubeslice"
+	if m.Options.Prefix != "" {
+		ns = m.Options.Prefix
```

Comment:
instead of setting the default ns in every method, you can set it in `NewMetricsFactory` method

Reply:
@kmjayadeep I made the change. Can you please re-review once?